tests/test_content_manifest.py

```diff
@@ -152,29 +152,63 @@ def test_process_pip(default_request, default_toplevel_purl):
     assert cm._pip_data == expected_contents
 
 
+def test_process_gitsubmodule(default_request):
+    submodule_commit_id = "58c88e4952e95935c0dd72d4a24b0c44f2249f5b"
+    pkg = Package.from_json(
+        {
+            "name": "tour",
+            "type": "git-submodule",
+            "version": f"https://github.com/testrepo/tour.git#{submodule_commit_id}",
+        }
+    )
+    pkg.id = 1
+    expected_purl = "pkg:github/testrepo/tour@58c88e4952e95935c0dd72d4a24b0c44f2249f5b"
+
+    cm = ContentManifest(default_request)
+
+    # emulate to_json behavior to setup internal packages cache
+    cm._gitsubmodule_data.setdefault(
+        pkg.id, {"purl": expected_purl, "dependencies": [], "sources": []}
+    )
+
+    expected_contents = {pkg.id: {"purl": expected_purl, "dependencies": [], "sources": []}}
+
+    assert cm._gitsubmodule_data
+    assert pkg.id in cm._gitsubmodule_data
+    assert cm._gitsubmodule_data == expected_contents
```

Comment:
Oh my bad, there's no `process_gitsubmodule_package()` method so there's nothing to test here. This test can be removed